dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresFilterTranslator.cs

```diff
@@ -0,0 +1,332 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Microsoft.SemanticKernel.Connectors.Postgres;
+
+internal class PostgresFilterTranslator
+{
+    private IReadOnlyDictionary<string, string> _storagePropertyNames = null!;
+    private ParameterExpression _recordParameter = null!;
+
+    private readonly List<object> _parameterValues = new();
+    private int _parameterIndex;
+
+    private readonly StringBuilder _sql = new();
+
+    internal (string Clause, List<object> Parameters) Translate(
```

Comment:
do you think there's room for SQL injection, consdiering that the result is injected into a SQL statement?

Reply:
Absolutely (and that's very important). There's [a test](https://github.com/microsoft/semantic-kernel/pull/10273/files#diff-54e49edd3099d4031761dc5017c0e6bf491e6b334de33f300a54fa1ac3269ee6R24) exercising an input with a single quote to check exactly for this. I have a note to do a general comprehensive pass to make sure everything is airtight - stuff like property names isn't currently safe (those generally aren't user input, but I still need to make them airtight).